File: app/src/main/java/com/ocram/qichwadic/presentation/ui/activity/SplashActivity.java
package com.ocram.qichwadic.presentation.ui.activity;

import android.content.Intent;
import android.os.Bundle;
import android.os.Handler;


import androidx.appcompat.app.AppCompatActivity;

import com.ocram.qichwadic.QichwaDicApplication;
import com.ocram.qichwadic.R;
import com.ocram.qichwadic.framework.preferences.PreferencesHelper;


import javax.inject.Inject;

import io.reactivex.disposables.CompositeDisposable;

public class SplashActivity extends AppCompatActivity {

    private CompositeDisposable compositeDisposable;

    @Inject
    PreferencesHelper preferencesHelper;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        preferencesHelper = QichwaDicApplication.get(this).getAppComponent().getPreferencesHelper();
        super.onCreate(savedInstanceState);
        this.compositeDisposable = new CompositeDisposable();
        setContentView(R.layout.activity_splash);
    }

    @Override
    protected void onResume() {
        super.onResume();
        Thread t = new Thread(() -> {
            compositeDisposable.add(
                    preferencesHelper.isFirstStart().subscribe(isFirstStart -> {
                        final Intent intent;
                        if(isFirstStart){
                            intent = new Intent(SplashActivity.this, IntroActivity.class);
                            preferencesHelper.saveFirstStart();
                        }else{
                            intent = new Intent(SplashActivity.this, SearchActivity.class);
                        }
                        runOnUiThread(() -> {
                            startActivity(intent);
                            overridePendingTransition(R.anim.enteranim, R.anim.exitanim);
                            SplashActivity.this.finish();
                        });
                    })
            );
        });
        new Handler().postDelayed(t::start, 1500);
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        if(this.compositeDisposable != null) {
            this.compositeDisposable.dispose();
        }
    }
}

File: app/src/main/java/com/ocram/qichwadic/features/intro/ui/SplashActivity.kt
package com.ocram.qichwadic.features.intro.ui

import android.content.Intent
import android.os.Bundle


import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.Observer

import com.ocram.qichwadic.R
import com.ocram.qichwadic.core.ui.activity.MainActivity
import org.koin.androidx.viewmodel.ext.android.viewModel
import java.util.*
import kotlin.concurrent.schedule


class SplashActivity : AppCompatActivity() {

    private val splashViewModel: SplashViewModel by viewModel()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_splash)
        splashViewModel.firstStart.observe(this, Observer { onFirstStartLoaded(it) })
    }

    private fun onFirstStartLoaded (isFirstStart: Boolean) {
        Timer("", false).schedule(1500) {
            val targetCls = if (isFirstStart) IntroActivity::class.java else MainActivity::class.java
            startActivity(Intent(this@SplashActivity, targetCls))
            overridePendingTransition(R.anim.enteranim, R.anim.exitanim)
            this@SplashActivity.finish()
        }

    }
}